packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -329,84 +354,141 @@ export default class CallDiagnosticMetrics extends StatelessWebexPlugin {
-          fields: {
-            meetingId,
-            name,
-          },
-        });
+      return undefined;
+    }
 
-        return;
+    // grab identifiers
+    const identifiers = this.getIdentifiers({
+      meeting,
+      mediaConnections: meeting?.mediaConnections || mediaConnections,
+    });
+
+    // check if we need to generate errors
+    const errors: ClientEvent['payload']['errors'] = [];
+
+    if (rawError) {
+      const generatedError = this.generateClientEventErrorPayload(rawError);
+      if (generatedError) {
+        errors.push(generatedError);
       }
+    }
 
-      // grab identifiers
-      const identifiers = this.getIdentifiers({
-        meeting,
-        mediaConnections: meeting?.mediaConnections || mediaConnections,
-      });
+    // create client event object
+    const clientEventObject: ClientEvent['payload'] = {
+      name,
+      canProceed: true,
+      identifiers,
+      errors,
+      eventData: {
+        webClientDomain: window.location.hostname,
+      },
+      userType: meeting.getCurUserType(),
+      loginType: this.getCurLoginType(),
+    };
 
-      // check if we need to generate errors
-      const errors: ClientEvent['payload']['errors'] = [];
+    return clientEventObject;
+  }
 
-      if (rawError) {
-        const generatedError = this.generateClientEventErrorPayload(rawError);
-        if (generatedError) {
-          errors.push(generatedError);
-        }
-      }
+  /**
+   * Create client event object for pre meeting events
+   * @param event - event key
+   * @param options - payload
+   * @returns object
+   */
+  private createClientEventObjectPreMeeting({
+    name,
+    options,
+  }: {
+    name: ClientEvent['name'];
+    options: SubmitClientEventOptions;
+  }) {
+    const {correlationId} = options;
+
+    // grab identifiers
+    const identifiers = this.getIdentifiers({
+      correlationId,
+    });
+
+    // create client event object
+    const clientEventObject: ClientEvent['payload'] = {
+      name,
+      canProceed: true,
+      identifiers,
+      eventData: {
+        webClientDomain: window.location.hostname,
+      },
+      loginType: this.getCurLoginType(),
+    };
 
-      // create client event object
-      let clientEventObject: ClientEvent['payload'] = {
-        name,
-        canProceed: true,
-        identifiers,
-        errors,
-        eventData: {
-          webClientDomain: window.location.hostname,
-        },
-        userType: meeting.getCurUserType(),
-        loginType: meeting.getCurLoginType(),
-      };
+    return clientEventObject;
+  }
 
-      // merge any new properties, or override existing ones
-      clientEventObject = merge(clientEventObject, payload);
+  /**
+   * Submit Client Event CA event.
+   * @param event - event key
+   * @param payload - additional payload to be merged with default payload
+   * @param options - payload
```

Comment:
this function has only one parameter which is an object, you have to reflect it in the JSDoc as well:
```suggestion
   * @param arg - submit parameters
   * @param arg.event - event key
   * @param arg.payload - additional payload to be merged with default payload
   * @param arg.options - payload
```
here `arg` could be anything else since you deconstruct the object in the function parameters.

Reply:
added